### ISSUE TITLE
ModuleNotFoundError: No module named 'setuptools_rust'

### ISSUE THREAD

**Issue by zmoog** [elastic/e2e-testing#2988]
Hi! 

In the last few days, I have had an E2E failing with the following error while the pipeline is busing [configuring a node for testing](https://github.com/elastic/e2e-testing/blob/main/.ci/Jenkinsfile#L560):

```text
i[2022-08-31T15:23:24.212Z] fatal: [3.135.249.43]: FAILED! => {"changed": false, "cmd": ["/bin/pip3", "install", "docker-compose"], "msg": "stdout: Collecting docker-compose
  Downloading https://files.pythonhosted.org/packages/f3/3e/ca05e486d44e38eb495ca60b8ca526b192071717387346ed1031ecf78966/docker_compose-1.29.2-py2.py3-none-any.whl (114kB)
Collecting dockerpty<1,>=0.4.1 (from docker-compose)
  Downloading https://files.pythonhosted.org/packages/8d/ee/e9ecce4c32204a6738e0a5d5883d3413794d7498fe8b06f44becc028d3ba/dockerpty-0.4.1.tar.gz
Requirement already satisfied: distro<2,>=1.5.0 in /usr/local/lib/python3.6/site-packages (from docker-compose)
Requirement already satisfied: requests<3,>=2.20.0 in /usr/lib/python3.6/site-packages (from docker-compose)
Collecting python-dotenv<1,>=0.13.0 (from docker-compose)
  Downloading https://files.pythonhosted.org/packages/30/5f/2e5c564bd86349fe6b82ca840f46acf6f4bb76d79ba9057fce3d3e008864/python_dotenv-0.20.0-py3-none-any.whl
Collecting docopt<1,>=0.6.1 (from docker-compose)
  Downloading https://files.pythonhosted.org/packages/a2/55/8f8cab2afd404cf578136ef2cc5dfb50baa1761b68c9da1fb1e4eed343c9/docopt-0.6.2.tar.gz
Requirement already satisfied: PyYAML<6,>=3.10 in /usr/lib64/python3.6/site-packages (from docker-compose)
Collecting docker[ssh]>=5 (from docker-compose)
  Downloading https://files.pythonhosted.org/packages/54/f3/7af47ead249fbb798d64a0438bad5c26f17ef6ac5cd324d802038eb10d90/docker-5.0.3-py2.py3-none-any.whl (146kB)
Collecting websocket-client<1,>=0.32.0 (from docker-compose)
  Downloading https://files.pythonhosted.org/packages/f7/0c/d52a2a63512a613817846d430d16a8fbe5ea56dd889e89c68facf6b91cb6/websocket_client-0.59.0-py2.py3-none-any.whl (67kB)
Collecting texttable<2,>=0.9.0 (from docker-compose)
  Downloading https://files.pythonhosted.org/packages/75/23/8170868d04b153b1b9ed1bb84348212bb4a08f31b292ef9d7f6ea648fd49/texttable-1.6.4-py2.py3-none-any.whl
Requirement already satisfied: jsonschema<4,>=2.5.1 in /usr/lib/python3.6/site-packages (from docker-compose)
Collecting cached-property<2,>=1.2.0; python_version < \"3.8\" (from docker-compose)
  Downloading https://files.pythonhosted.org/packages/48/19/f2090f7dad41e225c7f2326e4cfe6fff49e57dedb5b53636c9551f86b069/cached_property-1.5.2-py2.py3-none-any.whl
Requirement already satisfied: six>=1.3.0 in /usr/lib/python3.6/site-packages (from dockerpty<1,>=0.4.1->docker-compose)
Requirement already satisfied: chardet<3.1.0,>=3.0.2 in /usr/lib/python3.6/site-packages (from requests<3,>=2.20.0->docker-compose)
Requirement already satisfied: idna<2.8,>=2.5 in /usr/lib/python3.6/site-packages (from requests<3,>=2.20.0->docker-compose)
Requirement already satisfied: urllib3<1.25,>=1.21.1 in /usr/lib/python3.6/site-packages (from requests<3,>=2.20.0->docker-compose)
Collecting paramiko>=2.4.2; extra == \"ssh\" (from docker[ssh]>=5->docker-compose)
  Downloading https://files.pythonhosted.org/packages/04/e5/39ec73dd4a8769d6759b8d6c60a1b2c9337f585407c2ae8bfb8ccb734278/paramiko-2.11.0-py2.py3-none-any.whl (212kB)
Requirement already satisfied: cryptography>=2.5 in /usr/lib64/python3.6/site-packages (from paramiko>=2.4.2; extra == \"ssh\"->docker[ssh]>=5->docker-compose)
Collecting bcrypt>=3.1.3 (from paramiko>=2.4.2; extra == \"ssh\"->docker[ssh]>=5->docker-compose)
  Downloading https://files.pythonhosted.org/packages/99/f2/b71b9b5b2400fffac7d42c560ac89f302c4d8e328337b2f05f0a4d9e590d/bcrypt-4.0.0.tar.gz
    Complete output from command python setup.py egg_info:
    
            =============================DEBUG ASSISTANCE==========================
            If you are seeing an error here please try the following to
            successfully install cryptography:
    
            Upgrade to the latest pip and try again. This will fix errors for most
            users. See: https://pip.pypa.io/en/stable/installing/#upgrading-pip
            =============================DEBUG ASSISTANCE==========================
    
    Traceback (most recent call last):
      File \"<string>\", line 1, in <module>
      File \"/tmp/pip-build-0zq25bfd/bcrypt/setup.py\", line 11, in <module>
        from setuptools_rust import RustExtension
    ModuleNotFoundError: No module named 'setuptools_rust'
    
    ----------------------------------------

:stderr: WARNING: Running pip install with root privileges is generally not a good idea. Try `pip3 install --user` instead.
Command \"python setup.py egg_info\" failed with error code 1 in /tmp/pip-build-0zq25bfd/bcrypt/
"}
```

Here's a sample execution: 
* https://beats-ci.elastic.co/blue/organizations/jenkins/e2e-tests%2Fe2e-testing-mbp/detail/7.17/1087/pipeline/541

Looking at the [recent activity](https://beats-ci.elastic.co/blue/organizations/jenkins/e2e-tests%2Fe2e-testing-mbp/activity/), it seems like it's not happening with my branch only.

Do you have any advice on how to address this issue?

Thanks!

**Comment by v1v** [comment]
It happens for `7.17`

<img width="1205" alt="image" src="https://user-images.githubusercontent.com/2871786/192472421-8173911e-3f03-44cb-a6e1-068e7c448588.png">

<img width="904" alt="image" src="https://user-images.githubusercontent.com/2871786/192472529-7a2b3432-edff-497e-a4a0-e4e10612794c.png">


when running `make -C .ci setup-node` in `centos8_arm64`, `debian_10_arm64` 

```
[2022-09-26T13:49:11.449Z] TASK [geerlingguy.docker_arm : Install Docker Compose using Pip.] **************
[2022-09-26T13:49:22.688Z] fatal: [3.142.247.54]: FAILED! => {"changed": false, "cmd": ["/bin/pip3", "install", "docker-compose"], "msg": "stdout: Collecting docker-compose\n  Downloading https://files.pythonhosted.org/packages/f3/3e/ca05e486d44e38eb495ca60b8ca526b192071717387346ed1031ecf78966/docker_compose-1.29.2-py2.py3-none-any.whl (114kB)\nCollecting cached-property<2,>=1.2.0; python_version < \"3.8\" (from docker-compose)\n  Downloading [https://files.pythonhosted.org/packages/48/19/f2090f7dad41e225c7f2326e4cfe6fff49e57dedb5b53636c9551f86b069/cached_property-1.5.2-py2.py3-none-any.whl\nCollecting](https://files.pythonhosted.org/packages/48/19/f2090f7dad41e225c7f2326e4cfe6fff49e57dedb5b53636c9551f86b069/cached_property-1.5.2-py2.py3-none-any.whl/nCollecting) docker[ssh]>=5 (from docker-compose)\n  Downloading https://files.pythonhosted.org/packages/54/f3/7af47ead249fbb798d64a0438bad5c26f17ef6ac5cd324d802038eb10d90/docker-5.0.3-py2.py3-none-any.whl (146kB)\nCollecting python-dotenv<1,>=0.13.0 (from docker-compose)\n  Downloading [https://files.pythonhosted.org/packages/30/5f/2e5c564bd86349fe6b82ca840f46acf6f4bb76d79ba9057fce3d3e008864/python_dotenv-0.20.0-py3-none-any.whl\nRequirement](https://files.pythonhosted.org/packages/30/5f/2e5c564bd86349fe6b82ca840f46acf6f4bb76d79ba9057fce3d3e008864/python_dotenv-0.20.0-py3-none-any.whl/nRequirement) already satisfied: jsonschema<4,>=2.5.1 in /usr/lib/python3.6/site-packages (from docker-compose)\nCollecting websocket-client<1,>=0.32.0 (from docker-compose)\n  Downloading https://files.pythonhosted.org/packages/f7/0c/d52a2a63512a613817846d430d16a8fbe5ea56dd889e89c68facf6b91cb6/websocket_client-0.59.0-py2.py3-none-any.whl (67kB)\nCollecting docopt<1,>=0.6.1 (from docker-compose)\n  Downloading [https://files.pythonhosted.org/packages/a2/55/8f8cab2afd404cf578136ef2cc5dfb50baa1761b68c9da1fb1e4eed343c9/docopt-0.6.2.tar.gz\nRequirement](https://files.pythonhosted.org/packages/a2/55/8f8cab2afd404cf578136ef2cc5dfb50baa1761b68c9da1fb1e4eed343c9/docopt-0.6.2.tar.gz/nRequirement) already satisfied: requests<3,>=2.20.0 in /usr/lib/python3.6/site-packages (from docker-compose)\nRequirement already satisfied: PyYAML<6,>=3.10 in /usr/lib64/python3.6/site-packages (from docker-compose)\nRequirement already satisfied: distro<2,>=1.5.0 in /usr/local/lib/python3.6/site-packages (from docker-compose)\nCollecting dockerpty<1,>=0.4.1 (from docker-compose)\n  Downloading [https://files.pythonhosted.org/packages/8d/ee/e9ecce4c32204a6738e0a5d5883d3413794d7498fe8b06f44becc028d3ba/dockerpty-0.4.1.tar.gz\nCollecting](https://files.pythonhosted.org/packages/8d/ee/e9ecce4c32204a6738e0a5d5883d3413794d7498fe8b06f44becc028d3ba/dockerpty-0.4.1.tar.gz/nCollecting) texttable<2,>=0.9.0 (from docker-compose)\n  Downloading [https://files.pythonhosted.org/packages/75/23/8170868d04b153b1b9ed1bb84348212bb4a08f31b292ef9d7f6ea648fd49/texttable-1.6.4-py2.py3-none-any.whl\nCollecting](https://files.pythonhosted.org/packages/75/23/8170868d04b153b1b9ed1bb84348212bb4a08f31b292ef9d7f6ea648fd49/texttable-1.6.4-py2.py3-none-any.whl/nCollecting) paramiko>=2.4.2; extra == \"ssh\" (from docker[ssh]>=5->docker-compose)\n  Downloading https://files.pythonhosted.org/packages/04/e5/39ec73dd4a8769d6759b8d6c60a1b2c9337f585407c2ae8bfb8ccb734278/paramiko-2.11.0-py2.py3-none-any.whl (212kB)\nRequirement already satisfied: six in /usr/lib/python3.6/site-packages (from websocket-client<1,>=0.32.0->docker-compose)\nRequirement already satisfied: chardet<3.1.0,>=3.0.2 in /usr/lib/python3.6/site-packages (from requests<3,>=2.20.0->docker-compose)\nRequirement already satisfied: idna<2.8,>=2.5 in /usr/lib/python3.6/site-packages (from requests<3,>=2.20.0->docker-compose)\nRequirement already satisfied: urllib3<1.25,>=1.21.1 in /usr/lib/python3.6/site-packages (from requests<3,>=2.20.0->docker-compose)\nCollecting pynacl>=1.0.1 (from paramiko>=2.4.2; extra == \"ssh\"->docker[ssh]>=5->docker-compose)\n  Downloading https://files.pythonhosted.org/packages/a7/22/27582568be639dfe22ddb3902225f91f2f17ceff88ce80e4db396c8986da/PyNaCl-1.5.0.tar.gz (3.4MB)\nCollecting bcrypt>=3.1.3 (from paramiko>=2.4.2; extra == \"ssh\"->docker[ssh]>=5->docker-compose)\n  Downloading [https://files.pythonhosted.org/packages/99/f2/b71b9b5b2400fffac7d42c560ac89f302c4d8e328337b2f05f0a4d9e590d/bcrypt-4.0.0.tar.gz\n](https://files.pythonhosted.org/packages/99/f2/b71b9b5b2400fffac7d42c560ac89f302c4d8e328337b2f05f0a4d9e590d/bcrypt-4.0.0.tar.gz/n)    Complete output from command python setup.py egg_info:\n    \n            =============================DEBUG ASSISTANCE==========================\n            If you are seeing an error here please try the following to\n            successfully install cryptography:\n    \n            Upgrade to the latest pip and try again. This will fix errors for most\n            users. See: https://pip.pypa.io/en/stable/installing/#upgrading-pip\n            =============================DEBUG ASSISTANCE==========================\n    \n    Traceback (most recent call last):\n      File \"<string>\", line 1, in <module>\n      File \"/tmp/pip-build-u1xrrkut/bcrypt/setup.py\", line 11, in <module>\n        from setuptools_rust import RustExtension\n    ModuleNotFoundError: No module named 'setuptools_rust'\n    \n    ----------------------------------------\n\n:stderr: WARNING: Running pip install with root privileges is generally not a good idea. Try `pip3 install --user` instead.\nCommand \"python setup.py egg_info\" failed with error code 1 in /tmp/pip-build-u1xrrkut/bcrypt/\n"}
[2022-09-26T13:49:22.690Z] 
[2022-09-26T13:49:22.690Z] PLAY RECAP *********************************************************************
[2022-09-26T13:49:22.690Z] 3.142.247.54               : ok=11   changed=6    unreachable=0    failed=1    skipped=20   rescued=0    ignored=0   
[2022-09-26T13:49:22.690Z] localhost                  : ok=1    changed=0    unreachable=0    failed=0    skipped=0    rescued=0    ignored=0   
[2022-09-26T13:49:22.690Z] 
[2022-09-26T13:49:22.690Z] Header doesn't match the format: ********.
[2022-09-26T13:49:22.892Z] make: *** [Makefile:150: setup-node] Error 2
[2022-09-26T13:49:22.893Z] make: Leaving directory '/var/lib/jenkins/workspace/e2e-tests_e2e-testing-mbp_7.17/src/github.com/elastic/e2e-testing/.ci'
```

**Comment by zmoog** [comment]
Hey @pazone, thank you for working on this issue! 

Are these changes available to the E2E infra? I tried to run a `/test` on [one](https://github.com/elastic/beats/pull/33023) of my PRs, and I see some errors related to `setuptools_rust`.

**Comment by zmoog** [comment]
Hey @pazone, I'm sorry to bother you again, but I have the feeling I am missing some details here.

I still have two PRs on the 7.17 branch failing due to `ModuleNotFoundError: No module named 'setuptools_rust'`  error:

- https://github.com/elastic/beats/pull/32297
- https://github.com/elastic/beats/pull/33023

Any idea about what I am doing wrong?

Thanks! 🙇

**Comment by pazone** [comment]
@zmoog Thank for reporting this. I just need to backport the PR to the `7.17` branch

**Comment by pazone** [comment]
@zmoog the backport pull request has been merged to the 7.17 branch

**Comment by zmoog** [comment]
Hey @pazone, thank you so much! I'm running the tests again.
